cli/util/upload_bin.go

```diff
@@ -223,21 +222,21 @@ func fetchLatestMissingBlockIndex(ctx context.Context, p *pool.Pool, containerID
 					objectIDs, err = neofs.ObjectSearch(ctx, p, priv, containerID.String(), prm)
 					return err
 				}, maxRetries)
-				results[i] = searchResult{startIndex: startIndex, endIndex: endIndex, numOIDs: len(objectIDs), err: err}
-			}(i, startIndex, endIndex)
+				results[i] = searchResult{batchStartIndex: startIndex / uploadBatchSize, index: startIndex, numOIDs: len(objectIDs), err: err}
+			}(i, startIndex)
 		}
 		wg.Wait()
 
 		for i := len(results) - 1; i >= 0; i-- {
 			if results[i].err != nil {
-				return 0, fmt.Errorf("blocks search failed for batch with indexes from %d to %d: %w", results[i].startIndex, results[i].endIndex-1, results[i].err)
+				return 0, fmt.Errorf("blocks search failed for batch starting with index %d, block index %d: %w", results[i].batchStartIndex, results[i].index, results[i].err)
 			}
 			if results[i].numOIDs == 0 {
 				emptyBatchFound = true
 				continue
 			}
 			if emptyBatchFound || (batch == numBatches && i == len(results)-1) {
-				return results[i].startIndex, nil
+				return results[i].batchStartIndex * uploadBatchSize, nil
```

Comment:
It works improperly. Outer loop `for batch := numBatches; batch > -maxParallelSearches; batch -= maxParallelSearches {` is built based on the fact that every element of `results` is a result of 10K-blocks batch, exactly the same size as we use to upload batches. We need to check that there's no gaps within the batch of `uploadBatchSize` size.

Reply:
So now we have `numBatches = currentHeight`, and `batchStartIndex = index/uploadBatchSize`.  So once we have found the first not empty `results[i]` we can say that this is the last batch (calculated based on `uploadBatchSize`) and it can be not full, so we can return the `batchStartIndex * uploadBatchSize` - which will be end index of last uploaded batch.

Reply:
OK, will work, but needs to be refactored a bit. I want to make this code generic so that it works for both single-object search and multy-object search.